### PR TITLE
XWIKI-19352: Indexing of the attachments usage on the wiki

### PR DESCRIPTION
**Pull request by manuelleduc**
Wait for the document analysis tasks to be finished before continuing.

### CODE REVIEW

File: xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-data/pom.xml [hunk 1]
```diff
@@ -161,6 +161,12 @@
               <artifactId>xwiki-platform-ratings-api</artifactId>
               <version>${project.version}</version>
             </dependency>
+            <!-- Used to process the document analysis tasks (e.g., backlinks indexing). -->
+            <dependency>
```
Comment (tmortagne) [inline xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-data/pom.xml:165]: Not really needed here since it's already a dependency of the plugin.

Comment (manuelleduc) [reply xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-data/pom.xml:165]: Indeed, removing it, thanks.

File: xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java [hunk 2]
```diff
@@ -71,4 +75,40 @@ public void executeInternal() throws MojoExecutionException
...
+    private void consumeIndexingQueue() throws MojoExecutionException
+    {
+        ComponentManager componentManager =
+            (ComponentManager) this.oldCoreHelper.getXWikiContext().get(ComponentManager.class.getName());
+        try {
+            // Manually start the consumer threads as it is not automatically started in the packaging setup.
+            // This is also convenient as we can be sure that the tasks are all queued before being consumed.
+            TaskManager taskManager = componentManager.getInstance(TaskManager.class);
+            if (taskManager instanceof DefaultTasksManager) {
+                ((DefaultTasksManager) taskManager).startThread();
```
Comment (tmortagne) [inline xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java:100]: Why is this needed ? I see that this is done automatically in TaskApplicationReadyListener when receiving ApplicationReadyEvent and ApplicationReadyEvent seems to be sent in OldCoreHelper#create.

File: xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java [hunk 3]
```diff
@@ -71,4 +75,40 @@ public void executeInternal() throws MojoExecutionException
...
+            }
+
+            getLog().info("Waiting for the document indexing queue to be empty...");
```
Comment (tmortagne) [inline xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java:103]: Maybe indicate how big is the queue before we start to wait for it to empty (to give a clue of how long it might take).

File: xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java (hunk 2 above)
Comment (manuelleduc) [reply xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java:100]: `org.xwiki.index.internal.TaskApplicationReadyListener#initialize` is called but the context provider returns null. 
`org.xwiki.index.internal.TaskApplicationReadyListener#onEvent` is never called, which not be right.

Comment (tmortagne) [reply xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java:100]: > org.xwiki.index.internal.TaskApplicationReadyListener#initialize is called but the context provider returns null.

This is expected yes, same as in a standard XWiki instance.

>  org.xwiki.index.internal.TaskApplicationReadyListener#onEvent is never called, which not be right.

That's really not expected according to https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/org/xwiki/tool/utils/OldCoreHelper.java#L152.

Comment (manuelleduc) [reply xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java:100]: Alright, debugging.

Comment (tmortagne) [reply xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java:100]: > That's really not expected according to [https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/org/xwiki/tool/utils/OldCoreHelper.java#L152](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/org/xwiki/tool/utils/OldCoreHelper.java?rgh-link-date=2022-03-21T15%3A40%3A31Z#L152).

Actually this is ApplicationStartedEvent sorry and indeed we don't seems to send ApplicationReadyEvent.
